microsoft/PowerToys
# [FancyZones] Window switch shortcut fix

## Summary of the Pull Request

* Split ZoneSet into two classes
* Made sure that the zone-window map is updated when the window is reopened
* Updated unit tests


## PR Checklist

- [x] **Closes:** #16620


## Detailed Description of the Pull Request / Additional comments


## Validation Steps Performed

* Assign 2+ windows to the same zone
* Verify that windows are rotated when pressing the shortcut
* Close and reopen any window (it should be opened in a zone, so check the `Move newly created windows to their last known zone` option to be turned on)
* Verify rotation still works
